### ISSUE TITLE
Error data structures have unnecessary duplication of error codes

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#545]
The error encountered by @ebachoura in #363 was tracked down to a replication of an error code with the key in the object that holds it, e.g.:

https://github.com/dondi/GRNsight/blob/8fcaf47a0f319cba771515d02e032c5f840f985e/server/controllers/constants.js#L10-L57

Further, in other error dictionaries, this key to code does _not_ correspond, which also does not make sense for different reasons:

https://github.com/dondi/GRNsight/blob/8fcaf47a0f319cba771515d02e032c5f840f985e/server/controllers/spreadsheet-controller.js#L23-L115

(note also that in the example above, the data structure _sometimes_ points to an object and sometimes points to a function)

All told, this is a code smell: we should...
* Eliminate the duplication of keys/codes
* Ensure that the keys and codes always match
* Standardize the way these errors are invoked (i.e., object vs. functions)

**Comment by kdahlquist** [comment]
It seems that this is related to what @afiller1 is doing now.

**Comment by Onariaginosa** [comment]
When running GRNsight locally while running tests in another terninal,  error messages popped up within the localhost terminal. It appears that when I switched branches from beta to Onariaginosa, my dependencies were altered causing the error messages. I then deleted and reinstalled the dependencies, and the error messages did not reappear.

**Comment by dondi** [comment]
Inclusive in this refactor is the possibility of consolidating code into a distinct errors/warnings file, to be imported/required by other files as needed.

**Comment by Onariaginosa** [comment]
I consolidated all of the error and warnings to the existing `constants.js` file in server/controllers. The file originally had SIF specific warnings and errors in the file. I later noticed that there is a separate `graphml-constants.js` file in the same location. Should these two files be merged, or should I separate the excel and SIF warnings/errors into `excel-constants.js` and `sif-constants.js`?

**Comment by dondi** [comment]
Following the direction that `graphml-constant.js` was going, @dondi agrees that the more specific separation makes sense. Let’s just rename `excel-constants.js` to `workbook-constants.js`—sounds more consistent in terms of the format involved.

**Comment by Onariaginosa** [comment]
While I was in the process of splitting the files and renaming their references, I noticed that there were some GraphML related files that called the SIF constants. After looking closely I realised that the original constants.js included both SIF constants and general purpose constants. I decided to split the SIF constants and the general purpose constants into 2 files. Currently we have 4 total constant files:
- `graphml-constants.js`
- `workbook-constants.js`
- `sif-constants.js`
- `constants.js`

**Comment by Onariaginosa** [comment]
Waiting on issue #570 to do a pull request because many files changed/deleted were done in accordance with that issue.

**Comment by dondi** [comment]
QA review for @ahmad00m would be primarily regression testing: upload some workbooks with known errors/warnings and confirm that the appropriate messages appear. @Onariaginosa can provide guidance on what sheets to try.

**Comment by ahmad00m** [comment]
@ahmad00m checked out the importers files and the only duplication found was the numbers to letters object in additional-sheet-parser.js and workbook-constants.js

**Comment by dondi** [comment]
@Onariaginosa has the above issue addressed and will push a new PR once the current pending PR #920 has been merged.

**Comment by Onariaginosa** [comment]
Fixed issue in PR #925

**Comment by ahmad00m** [comment]
@ahmad00m confirms this issue is resolved.
